### PR TITLE
UCX: Option to pre-register communication buffer to populate the rcache.

### PR DESCRIPTION
**Pull request by bureddy**

### CODE REVIEW

File: src/team_lib/ucx/xccl_ucx_lib.c [hunk 1]
```diff
@@ -99,6 +99,12 @@ static ucs_config_field_t xccl_tl_ucx_context_config_table[] = {
      UCS_CONFIG_TYPE_UINT
     },
 
+    {"PRE_MEM_MAP", "0",
+     "register communication buffer with UCX duing collective init to populate registration cache",
```
Comment (vspetrov) [inline src/team_lib/ucx/xccl_ucx_lib.c]: duing -> during

File: src/team_lib/ucx/xccl_ucx_lib.c [hunk 2]
```diff
@@ -108,12 +114,60 @@ static ucs_config_field_t xccl_tl_ucx_context_config_table[] = {
     {NULL}
 };
 
+static size_t xccl_ucx_get_src_buffer_len(xccl_coll_op_args_t *coll_args, int team_size)
```
Comment (vspetrov) [inline src/team_lib/ucx/xccl_ucx_lib.c]: maybe have 1 implementation for both xccl_ucx_get_src/dst buffer_len to reduce code duplication? (the fn would have 1 more argument is_src ? )

File: src/team_lib/ucx/xccl_ucx_lib.c (hunk 1 above)
Comment (vspetrov) [inline src/team_lib/ucx/xccl_ucx_lib.c]: so, if "1" then src/dst buffers are registered during coll_init. And if "2" - all user mallocs are registered?

**Review by vspetrov — CHANGES_REQUESTED** [review]

File: src/team_lib/ucx/xccl_ucx_lib.c (hunk 1 above)
Comment (bureddy) [reply src/team_lib/ucx/xccl_ucx_lib.c]: yes

**Comment by bureddy** [comment]
@Sergei-Lebedev can you review and merge?

**Review by Sergei-Lebedev — APPROVED** [review]

**Pull request merged**